### DISCUSSION TITLE
Why there are two docker builders?

### DISCUSSION THREAD

**Discussion by lgxz** [aquasecurity/tracee#406] [Questions and Help] [answered]
1. golang:1.15-buster in Dockerfile.builder
2. golang:alpine in Dockerfile

Thanks!

**Comment by yanivagman** [comment, marked as answer]
Hi @lgxz,

As you can understand from the names, Dockerfile builds an image based on alpine while Dockerfile.builder doesn't.
For our image located in github we use the alpine based image, while for local build (DOCKER=1) we use the non alpine one.

The alpine build has a smaller image size, but building tracee locally using DOCKER=1 produces a binary that can be used locally without any container. If we would have used the alpine image to build locally, that would have create a dependency in the produced executable in musl-libc, which is used in alpine by default. As we don't want to create such a dependency (most people don't have musl installed locally), we are using a different docker builder for this purpose

**Comment by lgxz** [comment]
Thanks very much! This confused me a while.

By the way, It seems that "make bpf DOCKER=1" always build the bpf object along with tracee, not only the bpf object.
